[유동헌] #4 feat: add get clinic list function and test

dhhyy opened the pull request:
## 수정 사항 간략한 한줄 요약
- 조회 기능 구현
- 조회 기능 테스트 코드 작성

## 수정 사항들 자세한 내용


## 기타 특이 사항

dhhyy committed:
chore: create migrations file

- 마이그레이션 파일 생성

dhhyy committed:
feat: create clinic list view, test code

- clinic views.py 작성
- clinic tests.py 작성
- 리베이스 컴플릭트 해결

TedJunny commented on .gitignore:
```diff
@@ -357,3 +357,5 @@ zsdoc/data
 !/tests/_output/.gitkeep
 
 my_settings.py
+temp_db.csv
+db_uploader.py
```
.gitignore 파일은 PR에서 빼주세요

dhhyy replied:
.gitignore 수정하고 다시 올리겠습니다.

TedJunny commented on clinics/migrations/0002_alter_clinic_id.py:
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.9 on 2021-11-15 12:54
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('clinics', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='clinic',
+            name='id',
+            field=models.CharField(db_index=True, max_length=16, primary_key=True, serialize=False),
+        ),
+    ]
```
마이그레이션 파일은 어떤 부분 수정하신건가요?? 마이그레이션 파일은 개인이 수정해서 올리면 안되고 모델 수정할 것이 있으면 논의해서 올리겠습니다

dhhyy replied:
실수로 makemigrations 진행했습니다. 삭제하고 다시 올리겠습니다.

TedJunny commented on clinics/views.py:
```diff
@@ -1,3 +1,40 @@
+            clinics = Clinic.objects.all()[start:end]
+            
+            result = {
+                "currentCount" : len(clinics),
+                "data" : [
+                    {
+                        "name" : clinic.name,
+                        "id" : clinic.id,
+                        "duration" : clinic.duration,
+                        "scope" : clinic.scope,
+                        "type" : clinic.type,
+                        "institution" : clinic.institution,
+                        "trial" : clinic.trial,
+                        "subjects" : clinic.subjects,
+                        "department" : clinic.department
+                    } for clinic in clinics
+                ],
+                "matchCount" : len(Clinic.objects.all()),
+                "page" : page_num,
+                "perPage" : len(clinics),
+                "totalCount" : len(Clinic.objects.all())
+            }
```
key는 Camel Case가 아닌 Snake case로 맞춰주세요!

dhhyy replied:
네 수정하겠습니다.

TedJunny commented:
지윤님 PR이 메인에 합쳐져서 리베이스 후 충돌 해결 후 다시 PR 올려주세요

TedJunny reviewed (changes requested):
지윤님 PR이 메인에 합쳐져서 리베이스 후 충돌 해결 후 다시 PR 올려주세요

dhhyy committed:
chore: merge with main

Merge branch 'main' of https://github.com/wanted-InfinityLoop/human-scape-InfinityLoop into main
